docs/protocol/specifications/vm1.md

```diff
@@ -1100,44 +1104,59 @@ Note: [standard memory cost](#standard-memory-cost) is applied *after* the instr
+0xc1  | (index datahash amount assetid version prog → q)     | 16; [standard memory cost](#standard-memory-cost)
 
-1. Pops 6 items from the data stack: `index`, `refdatahash`, `amount`, `assetid`, `version`, `prog`.
+1. Pops 6 items from the data stack: `index`, `datahash`, `amount`, `assetid`, `version`, `prog`.
 2. Fails if `index` is negative or not a valid [number](#vm-number).
 3. Fails if the number of outputs is less or equal to `index`.
 4. Fails if `amount` and `version` are not non-negative [numbers](#vm-number).
-5. Finds a transaction output at the given `index`.
-6. If the output satisfies all of the following conditions pushes [true](#vm-boolean) on the data stack; otherwise pushes [false](#vm-boolean):
-    1. control program equals `prog`,
-    2. VM version equals `version`,
-    3. asset ID equals `assetid`,
-    4. amount equals `amount`,
-    5. `refdatahash` is an empty string or it matches the [SHA3-256](data.md#sha3) hash of the reference data.
+5. If the current entry is a [Mux](blockchain.md#mux-1):
+    1. Finds a [destination entry](blockchain.md#value-destination-1) at the given `index`.
+    2. If the entry satisfies all of the following conditions pushes [true](#vm-boolean) on the data stack; otherwise pushes [false](#vm-boolean):
+        1. the destination entry is an [output](blockchain.md#output-1) or a [retirement](blockchain.md#retirement-1),
+        2. if the destination is an output: control program equals `prog` and VM version equals `version`,
+        3. if the destination is a retirement: `prog` is an empty string and `version` is zero,
+        4. asset ID equals `assetid`,
+        5. amount equals `amount`,
+        6. `datahash` is an empty string or it matches the [SHA3-256](blockchain.md#sha3) hash of the data.
+5. If the entry is an [issuance](blockchain.md#issuance-1) or a [spend](blockchain.md#spend-1):
+    1. If the [destination entry](blockchain.md#value-destination-1) is a [Mux](blockchain.md#mux-1), performs checks as described in the step 5.
+    2. If the [destination entry](blockchain.md#value-destination-1) is an [output](blockchain.md#output-1) or a [retirement](blockchain.md#retirement-1):
+        1. If `index` is not zero, pushes [false](#vm-boolean) on the data stack.
+        2. Otherwise, performs checks as described in the step 5.2.
 
 Fails if executed in the [block context](#block-context).
 
+Fails if the entry is not a [mux](blockchain.md#mux-1), an [issuance](blockchain.md#issuance-1) or a [spend](blockchain.md#spend-1).
 
 #### ASSET
 
 Code  | Stack Diagram  | Cost
 ------|----------------|-----------------------------------------------------
-0xc2  | (∅ → assetid)   | 1; [standard memory cost](#standard-memory-cost)
+0xc2  | (∅ → assetid)  | 1; [standard memory cost](#standard-memory-cost)
+
+If the current entry is an [issuance](blockchain.md#issuance-1) or a [spend](blockchain.md#spend-1) entry, pushes the `Value.AssetID` of that entry.
```

Comment:
A spend doesn't have a `Value.AssetID`, but it does have a `SpentOutput.Source.Value.AssetID`.